shared/data/core/src/main/java/com/ivy/data/repository/TransactionRepository.kt

```diff
@@ -52,11 +55,50 @@ interface TransactionRepository {
         accountId: AccountId
     ): List<Transaction>
 
+    suspend fun findAllByCategoryAndTypeAndBetween(
+        categoryId: UUID,
+        type: TransactionType,
+        startDate: LocalDateTime,
+        endDate: LocalDateTime
+    ): List<Transaction>
+
+    suspend fun findAllUnspecifiedAndTypeAndBetween(
+        type: TransactionType,
+        startDate: LocalDateTime,
+        endDate: LocalDateTime
+    ): List<Transaction>
+
+    suspend fun findAllUnspecifiedAndBetween(
+        startDate: LocalDateTime,
+        endDate: LocalDateTime
+    ): List<Transaction>
+
+    suspend fun findAllByCategoryAndBetween(
+        categoryId: UUID,
+        startDate: LocalDateTime,
+        endDate: LocalDateTime
+    ): List<Transaction>
+
+    suspend fun flagDeletedByAccountId(accountId: UUID)
+
     suspend fun save(value: Transaction)
     suspend fun saveMany(value: List<Transaction>)
 
     suspend fun flagDeleted(id: TransactionId)
+    suspend fun flagDeletedByRecurringRuleIdAndNoDateTime(recurringRuleId: UUID)
     suspend fun deleteById(id: TransactionId)
     suspend fun deleteAllByAccountId(accountId: AccountId)
     suspend fun deleteAll()
+
+    suspend fun findLoanTransaction(
+        loanId: UUID
+    ): TransactionEntity?
+
+    suspend fun findLoanRecordTransaction(
+        loanRecordId: UUID
+    ): TransactionEntity?
+
+    suspend fun findAllByLoanId(
+        loanId: UUID
+    ): List<TransactionEntity>
```

Comment:
same here, `Transaction`. The outside world should be abstracted from Room DB entities

Reply:
Sure. 
Okay got it, so this was the reason.